File: amp-sdk/go/client/eip712_test.go
package client

import (
	"encoding/hex"
	"testing"
)

// Cross-language KAT: must match the Rust/C#/Python/JS digest helpers and the
// server's compute_outcome_eip712_digest
// (amp-server/src/main.rs::test_outcome_digest_known_vector_cross_lang).
func TestComputeOutcomeEip712DigestKnownVector(t *testing.T) {
	var zeroAddr [20]byte
	digest, err := ComputeOutcomeEip712Digest("1", 1, make([]byte, 32), 43113, zeroAddr[:])
	if err != nil {
		t.Fatalf("digest error: %v", err)
	}
	got := hex.EncodeToString(digest[:])
	want := "2d2525ad5098ca8f82a2a6cabc6775c40a55df96dfa2fbb46d7c0e372b99096c"
	if got != want {
		t.Fatalf("digest mismatch:\n got %s\nwant %s", got, want)
	}
}

func TestComputeOutcomeEip712DigestRejectsShortTranscript(t *testing.T) {
	var zeroAddr [20]byte
	if _, err := ComputeOutcomeEip712Digest("1", 1, make([]byte, 16), 43113, zeroAddr[:]); err == nil {
		t.Fatal("expected error for 16-byte transcript hash")
	}
}

File: amp-sdk/js/src/test/eip712.test.ts
/**
 * Cross-language KAT for the EIP-712 outcome digest.
 * Same vector as test_outcome_digest_known_vector_cross_lang in amp-server.
 */
import { test } from "node:test";
import assert from "node:assert/strict";
import { computeOutcomeEip712Digest } from "../eip712.js";

test("outcome digest matches cross-language KAT", () => {
  const expected =
    "2d2525ad5098ca8f82a2a6cabc6775c40a55df96dfa2fbb46d7c0e372b99096c";
  const d = computeOutcomeEip712Digest("1", 1, new Uint8Array(32));
  const hex = Buffer.from(d).toString("hex");
  assert.equal(
    hex,
    expected,
    "JS digest must match Rust/C#/Python — see test_outcome_digest_known_vector_cross_lang",
  );
});

test("outcome digest differs on different transcript", () => {
  const d1 = computeOutcomeEip712Digest("1", 1, new Uint8Array(32));
  const d2 = computeOutcomeEip712Digest("1", 1, new Uint8Array(32).fill(1));
  assert.notEqual(Buffer.from(d1).toString("hex"), Buffer.from(d2).toString("hex"));
});

test("outcome digest rejects invalid inputs", () => {
  assert.throws(() => computeOutcomeEip712Digest("1", 1, new Uint8Array(16)), /32 bytes/);
  assert.throws(() => computeOutcomeEip712Digest("1", 5, new Uint8Array(32)), /1\.\.=4/);
});
